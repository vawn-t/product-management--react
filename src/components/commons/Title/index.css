/* - Title */
/* - ---------------------------------------------- */
.title {
  color: var(--dark);
  font-size: 24px;
  font-weight: 400;
  letter-spacing: 0.01rem;
}

/* - Subtitle */
/* - ---------------------------------------------- */
.subtitle {
  color: var(--primary);
  font-size: 20px;
  font-weight: 300;
  letter-spacing: 0.01rem;
}

/* - Custom */
/* - ---------------------------------------------- */
.card-title {
  width: 100%;
  max-width: 14rem;
  line-height: 2rem;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}
