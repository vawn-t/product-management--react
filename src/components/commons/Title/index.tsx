// Libraries
import React, { memo, ReactNode } from 'react';

// Styles
import './index.css';

// Types
import { FwType, VariantTypes } from '@common-types/index';

export interface TitleProps {
  children: ReactNode;
  className?: string;
  color?: string;
  fontWeight?: FwType;
  fontSize?: string;
  fontStyle?: 'inherit' | 'italic';
  lineHeight?: string;
  variant?: VariantTypes;
}

const Title: React.FC<TitleProps> = ({
  children,
  className = '',
  color,
  fontWeight,
  fontSize,
  fontStyle = 'inherit',
  lineHeight,
  variant = VariantTypes.Default
}) => (
  <div
    className={`${variant} ${className}`.trim()}
    style={{
      fontSize: fontSize,
      color: color,
      fontStyle: fontStyle,
      fontWeight: fontWeight,
      lineHeight: lineHeight
    }}
  >
    {children}
  </div>
);

export default memo(Title);
