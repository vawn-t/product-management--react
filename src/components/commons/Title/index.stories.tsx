// Libraries
import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

// Components
import Title from './index';

// Constants

// Styles
import '@assets/styles/reset.css';
import '@assets/styles/App.css';
import '@assets/styles/variables.css';
import './index.css';
import { FwType, VariantTypes } from '@common-types/index';

export default {
  title: 'Components/Title',
  component: Title
} as ComponentMeta<typeof Title>;

const Template: ComponentStory<typeof Title> = (args) => <Title {...args} />;

export const Default = Template.bind({});
Default.args = {
  children: 'Title',
  variant: VariantTypes.Default
};

export const Subtitle = Template.bind({});
Subtitle.args = {
  children: 'Title',
  variant: VariantTypes.Subtitle,
  fontWeight: FwType.Bold,
  fontStyle: 'italic'
};
