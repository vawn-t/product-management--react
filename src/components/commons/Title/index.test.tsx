// Libraries
import React from 'react';
import { cleanup } from '@testing-library/react';
import { create } from 'react-test-renderer';

// Components
import Title, { TitleProps } from './index';

// Types
import { FwType, VariantTypes } from '@common-types/index';

describe('Title component', () => {
  let container: HTMLElement;
  beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement('div');
    document.body.appendChild(container);
  });

  afterEach(() => {
    // cleanup on exiting
    container.remove();
    jest.clearAllMocks();
    cleanup();
  });

  test('should render correctly', () => {
    const title = create(<Title>Title</Title>).toJSON();
    expect(title).toMatchSnapshot();

    const subtitle = create(
      <Title variant={VariantTypes.Subtitle}>Title</Title>
    ).toJSON();
    expect(subtitle).toMatchSnapshot();
  });

  test('should render with custom style', () => {
    const customProps: TitleProps = {
      color: '#0f0',
      fontWeight: FwType.Bold,
      fontSize: '24px',
      fontStyle: 'italic',
      children: 'Title'
    };

    const title = create(<Title {...customProps} />).toJSON();

    expect(title).toMatchSnapshot();
  });
});
