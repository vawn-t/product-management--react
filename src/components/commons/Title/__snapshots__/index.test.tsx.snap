// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title component should render correctly 1`] = `
<div
  className="title"
  style={
    Object {
      "color": undefined,
      "fontSize": undefined,
      "fontStyle": "inherit",
      "fontWeight": undefined,
      "lineHeight": undefined,
    }
  }
>
  Title
</div>
`;

exports[`Title component should render correctly 2`] = `
<div
  className="subtitle"
  style={
    Object {
      "color": undefined,
      "fontSize": undefined,
      "fontStyle": "inherit",
      "fontWeight": undefined,
      "lineHeight": undefined,
    }
  }
>
  Title
</div>
`;

exports[`Title component should render with custom style 1`] = `
<div
  className="title"
  style={
    Object {
      "color": "#0f0",
      "fontSize": "24px",
      "fontStyle": "italic",
      "fontWeight": 700,
      "lineHeight": undefined,
    }
  }
>
  Title
</div>
`;
